pkg/cmd/pulumi/operations/refresh.go

```diff
@@ -117,6 +120,15 @@ func NewRefreshCmd() *cobra.Command {
 				return err
 			}
 
+			usesTargets := opts.Engine.Targets.IsConstrained() || opts.Engine.TargetDependents
+			usesExcludes := opts.Engine.Excludes.IsConstrained() || opts.Engine.ExcludeDependents
```

Comment:
Do we need the same for preview?  I also wonder if we could use the [`MarkFlagsMutuallyExclusive`](https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsMutuallyExclusive) functionaly of cobra for this.